Dungeons/Dungeons/CorridorNodeLayouter.cs

```diff
@@ -14,134 +17,281 @@ class CorridorNodeLayouter : INodeLayouter
+      var width = max.Item1 + 1;
+      var height = max.Item2 + 1;
+      level.Create(width, height);
+
+      //var doors = localLevel.GetTiles<IDoor>();
+      //var secret = doors.Any(i => i.Secret);
+      //if(!info.PreventSecretRoomGeneration)
+      //  Debug.Assert(secret);
+      level.AppendMaze(localLevel, new Point(0, 0), new Point(width, height));
       level.DeleteWrongDoors();
+      //if (secret)
+      //{
+      //  var secretAppended = level.GetTiles<IDoor>().Any(i => i.Secret);
+      //  Debug.Assert(secretAppended);
+      //}
 
+      level.SecretRoomIndex = -1;
+      var sn = nodes.Where(i => i.Secret).FirstOrDefault();
+      if (sn != null)
+        level.SecretRoomIndex = sn.NodeIndex;
       return level;
     }
 
-    protected virtual void LayoutNodes(DungeonNode localLevel, List<DungeonNode> mazeNodes)
+    protected virtual void LayoutNodes(DungeonNode level, List<DungeonNode> mazeNodes)
     {
-      //AppendNodeInfo info = new AppendNodeInfo();
-      //info.side = EntranceSide.Right;
-      //float chanceForLevelTurn = 0.5f;
-      //EntranceSide? prevEntranceSide = null;
+      this.mazeNodes = mazeNodes;
+      var random = new Random();
 
-      //for (int nodeIndex = 0; nodeIndex < mazeNodes.Count; nodeIndex++)
-      //{
-      //  var infoNext = CalcNextValues(mazeNodes, info, chanceForLevelTurn, nodeIndex);
-      //  if (nodeIndex < mazeNodes.Count - 1 && generateLayoutDoors)
-      //    mazeNodes[nodeIndex].GenerateLayoutDoors(infoNext.side, mazeNodes[nodeIndex + 1]);
-
-      //  EntranceSide? entranceSideToSkip = null;
-      //  if (nodeIndex > 0)
-      //  {
-      //    if (prevEntranceSide == EntranceSide.Right)
-      //      entranceSideToSkip = EntranceSide.Left;
-      //    else if (prevEntranceSide == EntranceSide.Bottom)
-      //      entranceSideToSkip = EntranceSide.Top;
-      //    else
-      //      Debug.Assert(false);
-      //  }
-      //  if (forcedEntranceSideToSkip != null)
-      //    entranceSideToSkip = forcedEntranceSideToSkip.Value;
-
-      //  mazeNodes[nodeIndex].Reveal(options.RevealAllNodes, true);
-
-      //  localLevel.AppendMaze
-      //  (
-      //    mazeNodes[nodeIndex],
-      //    new Point(info.nextX, info.nextY),
-      //    null,
-      //    false,
-      //    entranceSideToSkip,
-      //    nodeIndex > 0 ? mazeNodes[nodeIndex - 1] : null
-      //  );
-
-      //  prevEntranceSide = infoNext.side;
-      //  info = infoNext;
-      //}
+      var currentWidth = 0;
+      var currentHeight = 0;
+      List<int> distances = new List<int>();
+      mazeNodes[0].GenerateLayoutDoorsCorrindor(EntranceSide.Bottom);
+      mazeNodes[0].GenerateLayoutDoorsCorrindor(EntranceSide.Right);
+      mazeNodes[1].GenerateLayoutDoorsCorrindor(EntranceSide.Top);
+      mazeNodes[1].GenerateLayoutDoorsCorrindor(EntranceSide.Right);
+      mazeNodes[2].GenerateLayoutDoorsCorrindor(EntranceSide.Left);
```

Comment:
` mazeNodes[1].GenerateLayoutDoorsCorrindor(EntranceSide.Top);
      mazeNodes[1].GenerateLayoutDoorsCorrindor(EntranceSide.Right);`
      
      To powinno byc w metodzie:
GenerateDoors(RoomPlacement roomPlacement,  EntranceSides[] entranceSide)
{

}